src/tbb/arena.cpp

```diff
@@ -395,9 +395,11 @@ bool arena::is_top_priority() const {
     return my_is_top_priority.load(std::memory_order_relaxed);
 }
 
-bool arena::try_join() {
+bool arena::try_join(bool should_join) {
     if (num_workers_active() < my_num_workers_allotted.load(std::memory_order_relaxed)) {
-        my_references += arena::ref_worker;
+        if (should_join) {
+            my_references += arena::ref_worker;
+        }
```

Comment:
That functions now looks contradictory to itself. It basically said "try join arena by adding reference, if possible", but now there is a boolean argument that says to not do so which basically acts like some sort of `joinable()` function. 

Following my previous comment, shouldn't we introduce `arena::joinable()` with following implementation:
```cpp
bool arena::joinable() {
    return num_workers_active() < my_num_workers_allotted.load(std::memory_order_relaxed);
}
```
Thus implementation of `try_join` would be the following:
```cpp
bool arena::try_join() {
    if (joinable()) {
        my_references += arena::ref_worker;
        return true;
    }
    return false;
}
```

Reply:
Makes sense. I added `arena::is_joinable()`.